### ISSUE TITLE
Restructure index files

### ISSUE THREAD

**Issue by rabitt** [mir-dataset-loaders/mirdata#277]
This will be a pain, but it's looking increasingly necessary - 

Index files are currently in the form:
```
{
    'track1-id': {...},
    'track2-id': {...}
...
}
```

This works fine for track-major datasets, but starts to break down for other types of datasets, such as relational or multitrack datasets. We'd also like to support [version information](#265) 

What about something like the following structure:
```
{
    'version': 1.0  # optional field, if it's not here we assume one version
    'tracks': {   # all the data we currently have in the index, when applicable
        'track1-id': {...},
        'track2-id': {...}
    },
    'multitracks': {  # optional field, when applicable
        'mtrack1-id': {
            'tracks': [track1-id, track5-id],    # map to the index of the tracks belonging to this multitrack
            'mix-audio': [path, asdfasdfaschecksum]   # store any multitrack-level annotations here like we do for tracks
        }
    }
    ...  # this leaves us room to add new fields for new top level data structures as we need them
} 
```

cc @magdalenafuentes

**Comment by magdalenafuentes** [comment]
Going through the different issues many have a common concern: should we make the `Track` object extremely flexible to support `multitrack` and `symbolic` data? Or should we standardize things more? Some thoughts: 


**why we want only a Track object**:

1. _simplify code_. If we add two more classes we will have to modify tests and probably repeat code in different scripts, and when we write the data generators as well. That makes me feel quite uneasy. 
2. _flexible enough_. So far it seems that most things we discussed (except the recommendation datasets #272) can be addressed by modifying `to_jams` and adding attributes.

**Why we want multiple objects depending on the dataset:**

1. _structure_. It could help give more structure to the library. It's the discussion we always have about forcing some structure but respecting the dataset idiosyncrasy. This seems to do both.
2. _clarity_. I wonder if is clearer to have multiple objects. 

Idea: should we put down in a list the datasets we plan to include, see the requirements of each, and then decide which changes to do on the API (objects, attributes, functions) based on #267 #282 #272 #276 ?

**Comment by magdalenafuentes** [comment]
We could also have #224, #225 and #233 in mind when we do so.

**Comment by magdalenafuentes** [comment]
Closed via #327
